src/models/krate.rs

```diff
@@ -528,6 +511,34 @@ impl CrateVersions for [Crate] {
     }
 }
 
+#[derive(Debug, Eq, PartialEq, thiserror::Error)]
+pub enum InvalidFeature {
+    #[error("feature cannot be empty")]
+    Empty,
+    #[error(
+        "invalid character `{0}` in feature `{1}`, the first character must be \
+        a Unicode XID start character or digit (most letters or `_` or `0` to \
+        `9`)"
+    )]
+    Start(char, String),
+    #[error(
+        "invalid character `{0}` in feature `{1}`, characters must be Unicode \
+        XID characters, `+`, `-`, or `.` (numbers, `+`, `-`, `_`, `.`, or most \
+        letters)"
+    )]
+    Char(char, String),
+    #[error(transparent)]
+    DependencyName(#[from] InvalidDependencyName),
```

Comment:
I'm uncertain about placing it here, as it is also used in validate_dependency. It's not intended for validating the feature name. But I can help to improve it in my next PR.

Reply:
InvalidDependencyName can also be used directly :)

Reply:
Oh. Got it. It's nice. Thanks!